packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -14,17 +13,80 @@ import 'package:flutter/scheduler.dart';
 ///
 /// ImageInfo objects are used by [ImageStream] objects to represent the
 /// actual data of the image once it has been obtained.
+///
+/// The receiver of an [ImageInfo] object must call [Image.dispose] on the
+/// [image]. To safely share the object with other clients, use the [clone]
+/// method.
 @immutable
 class ImageInfo {
   /// Creates an [ImageInfo] object for the given [image] and [scale].
   ///
-  /// Both the image and the scale must not be null.
+  /// Both the [image] and the [scale] must not be null.
   ///
-  /// The tag may be used to identify the source of this image.
+  /// The [debugLabel] may be used to identify the source of this image.
   const ImageInfo({ required this.image, this.scale = 1.0, this.debugLabel })
     : assert(image != null),
       assert(scale != null);
 
+  /// Creates an [ImageInfo] with a cloned [image].
```

Comment:
i think this class should have a dispose, since it owns something that can be disposed. That would be more intuitive than having to clean out its contents.

Reply:
The only reason I didn't do that is because we have some objects that eventually are just given the image out of an ImageInfo.

I think it's confusing either way, I'm just not sure which way is less confusing :)